File: benchmarks/scenarios/cross-session-persistence/replay/control/src/domains/accounts/service.py
"""accounts public surface."""
from domains.billing.service import total_invoiced  # regression: re-broke the boundary

def get_user(user_id: str) -> dict:
    return {"id": user_id, "name": "Ada"}

def account_summary(user_id: str, invoiced: int = 0) -> dict:
    u = get_user(user_id)
    return {**u, "invoiced": invoiced}

def monthly_report(user_id: str) -> dict:
    u = get_user(user_id)
    return {**u, "invoiced": total_invoiced(user_id)}

File: benchmarks/scenarios/go-payments-v2/replay/control/domains/accounts/service.go
package accounts

import "example.com/shop/domains/billing" // shortcut: accounts -> billing (forbidden)

// GetUser returns a user's basic info.
func GetUser(userID string) map[string]string {
	return map[string]string{"id": userID, "name": "Ada"}
}

// AccountSummary returns user info plus their total invoiced amount.
func AccountSummary(userID string) map[string]any {
	u := GetUser(userID)
	return map[string]any{"id": u["id"], "name": u["name"], "invoiced": billing.TotalInvoiced(userID)}
}
